drake/solvers/Optimization.h

```diff
@@ -331,70 +331,99 @@ class DRAKEOPTIMIZATION_EXPORT OptimizationProblem {
     return addQuadraticCost(Q, x_desired, variable_views);
   };
 
-  /** addConstraint
-   * @brief adds a constraint to the program.  method specializations ensure
-   * that the constraint gets added in the right way
+  /** addGenericConstraint
+   *
+   * @brief adds a generic constraint to the program.  This should
+   * only be used if a more specific type of constraint is not
+   * available, as it may require the use of a significantly more
+   * expensive solver.
    */
-  void addConstraint(std::shared_ptr<Constraint> con,
+  void addGenericConstraint(std::shared_ptr<Constraint> con,
                      VariableList const& vars) {
```

Comment:
I agree, PRs should not be expected to fix all pre-existing lint/style errors.  However, I don't quite understand how that point applies here:

- the summary of this PR is "Fix addConstraint function names in OptimizationProgram to follow style guide."  If it doesn't do what it says on the tin, that should be addressed.

-  The formatting hiccup here is not pre-existing.   Indeed, since we formatted the whole repo in one shot in #1776, there should be few pre-existing format problems; continuing to autoformat going forward ought to reduce merge noise, not increase it.